src/app/vertextool/qgsvertexeditor.cpp

```diff
@@ -350,49 +360,40 @@ void QgsVertexEditor::updateVertexSelection( const QItemSelection &selected, con
   mUpdatingVertexSelection = true;
 
   mSelectedFeature->deselectAllVertices();
-  Q_FOREACH ( const QModelIndex &index, mTableView->selectionModel()->selectedRows() )
+
+  QgsCoordinateTransform t( mLayer->crs(), mCanvas->mapSettings().destinationCrs(), QgsProject::instance() );
+  QgsRectangle *bbox = nullptr;
+  QModelIndexList indexList = selected.indexes();
+  for ( int i = 0; i < indexList.length(); ++i )
   {
-    int vertexIdx = index.row();
+    int vertexIdx = indexList.at( i ).row();
     mSelectedFeature->selectVertex( vertexIdx );
-  }
 
-  //ensure that newly selected vertex is visible in canvas
-  if ( !selected.indexes().isEmpty() )
-  {
-    int newRow = selected.indexes().first().row();
-    zoomToVertex( newRow );
+    // create a bounding box of selected vertices
+    QgsPointXY point( mSelectedFeature->vertexMap().at( vertexIdx )->point() );
+    if ( !bbox )
+      bbox = new QgsRectangle( point, point );
```

Comment:
I think that a `unique_ptr` could be used in this case (to avoid the manual `delete`).